### PR TITLE
NEXUS-7850: NEXUS-7851: Link validation

### PR DESCRIPTION
**Pull request by cstamas**
Link validation.

Issues
https://issues.sonatype.org/browse/NEXUS-7850
https://issues.sonatype.org/browse/NEXUS-7851

CI
http://bamboo.s/browse/NX-OSSF463

### CODE REVIEW

File: components/nexus-core/src/main/java/org/sonatype/nexus/proxy/storage/local/fs/DefaultFSLocalRepositoryStorage.java [hunk 1]
```diff
@@ -333,6 +339,33 @@ public void storeItem(Repository repository, StorageItem item)
...
+          final BufferedInputStream bufferedInputStream;
+          if (!cl.isReusable()) {
+            bufferedInputStream = new BufferedInputStream(cl.getContent());
+            bufferedInputStream.mark(1024);
+            // link persister with close the stream, so wrap it and prevent closing
+            cl = new PreparedContentLocator(new FilterInputStream(bufferedInputStream) {
+              @Override
+              public void close() {
+                // nop
+              }
```
Comment (mcculls) [inline components/nexus-core/src/main/java/org/sonatype/nexus/proxy/storage/local/fs/DefaultFSLocalRepositoryStorage.java]: You could consider moving this anonymous subclass to the buffered stream, ie:
```
            bufferedInputStream = new BufferedInputStream(cl.getContent()) {
              @Override
              public void close() {
                // nop
              }
            };
```
which would avoid the need for the `FilterInputStream` and the extra layer of indirection.

**Comment by mcculls** [comment]
+1 looks reasonable, tests to confirm the expected behaviour would be appreciated

File: components/nexus-core/src/main/java/org/sonatype/nexus/proxy/storage/local/fs/DefaultFSLocalRepositoryStorage.java (hunk 1 above)
Comment (cstamas) [inline components/nexus-core/src/main/java/org/sonatype/nexus/proxy/storage/local/fs/DefaultFSLocalRepositoryStorage.java]: Roger that, will change.

**Comment by cstamas** [comment]
UT added

**Comment by cstamas** [comment]
Manually merged as 8fd68e5be7ce56a1a618711f84e13fab754eb5a2

**Pull request closed without merging**